tests/ui/offset-of/offset-of-tuple-nested.rs

```diff
@@ -0,0 +1,32 @@
+// run-pass
+// Test for issue #112204 -- make sure this goes through the entire compilation pipeline,
+// similar to why `offset-of-unsized.rs` is also build-pass
+
+#![feature(offset_of)]
+#![feature(builtin_syntax)]
+
+use std::mem::offset_of;
+
+type ComplexTup = ((u8, (u8, (u8, u16), u8)), (u8, u32, u16));
+
+fn main() {
+    println!("{}", offset_of!(((u8, u8), u8), 0));
+    println!("{}", offset_of!(((u8, u8), u8), 1));
+    println!("{}", offset_of!(((u8, (u8, u8)), (u8, u8, u8)), 0.1.0));
+
+    // Complex case: do all combinations of spacings because the spacing determines what gets
+    // sent to the lexer.
+    println!("{}", offset_of!(ComplexTup, 0.1.1.1));
+    println!("{}", builtin # offset_of(ComplexTup, 0. 1.1.1));
```

Comment:
Note that this isn't supported by the `offset_of!()` macro's declaration:

```Rust
pub macro offset_of($Container:ty, $($fields:tt).+ $(,)?) {
    builtin # offset_of($Container, $($fields).+)
}
```

Does it make sense to change it to:

```Rust
pub macro offset_of($Container:ty, $($fields:tt)+) {
    builtin # offset_of($Container, $($fields)+)
}
```

and then let the builtin syntax deal with it?

This pattern *is* supported for normal tuples.

Reply:
Hmm yeah that's a bit complicated to do because it means that the builtin `offset_of` needs to get *way* better when it comes to errors... maybe it would be good for a follow up PR?

Reply:
If the `macro` arguments weren't somewhat important as documentation, I'd rather go with
```rust
macro offset_of($($tt:tt)+) { ... }
```
and do all the work in the parser.